base/src/Mp4WriterSinkUtils.cpp

```diff
@@ -83,24 +83,55 @@ void Mp4WriterSinkUtils::parseTSJpeg(uint64_t &ts, uint32_t &chunkTimeInMinutes,
 
 	// used cached values if the difference in ts is less than chunkTime
 	uint32_t chunkTimeInSecs = 60 * chunkTimeInMinutes;
-	if ((t - lastVideoTS) < chunkTimeInSecs && currentFolder == baseFolder)
+	if ((t - lastVideoTS) < chunkTimeInSecs && currentFolder == baseFolder && chunkTimeInMinutes != UINT32_MAX)
 	{
 		relPath = lastVideoFolderPath;
 		mp4FileName = lastVideoName;
 		nextFrameFileName = filePath(relPath, mp4FileName, baseFolder);
 		return;
 	}
 
-	// get new video path
-	std::string yyyymmdd = std::to_string(1900 + tm.tm_year) + format_2(tm.tm_mon) + format_2(tm.tm_mday);
-	relPath = boost::filesystem::path(yyyymmdd) / format_hrs(tm.tm_hour);
-	mp4FileName = std::to_string(ts) + ".mp4";
-
 	// cache new values
 	currentFolder = baseFolder;
 	lastVideoTS = t;
-	lastVideoFolderPath = relPath;
 	lastVideoMinute = tm.tm_min;
+
+	if (boost::filesystem::extension(baseFolder) == ".mp4")
```

Comment:
again same code block appears twice, once for JPEG, once for H264... consider DRY

Reply:
Resolved it. Added it in a function and calling that particular function now